parsers/IN_WE.py

```diff
@@ -0,0 +1,318 @@
+#!/usr/bin/env python3
+
+
+import json
+from datetime import datetime
+from logging import Logger, getLogger
+from typing import Optional
+
+import arrow
+import pandas as pd
+import pytz
+from requests import Session
+
+from parsers.lib.exceptions import ParserException
+
+IN_WE_PROXY = "https://in-proxy-jfnx5klx2a-el.a.run.app"
+PRODUCTION_URL = f"{IN_WE_PROXY}/GeneratorSchedule_data.aspx/Get_GeneratorScheduleData_state_Wise?host=https://www.wrldc.in"
+EXCHANGE_URL = f"{IN_WE_PROXY}/InterRegionalLinks_Data.aspx/Get_InterRegionalLinks_Region_Wise?host=https://www.wrldc.in"
+CONSUMPTION_URL = f"{IN_WE_PROXY}/OnlinestateTest1.aspx/GetRealTimeData_state_Wise?host=https://www.wrldc.in"
+
+POWER_PLANT_MAPPING = {
+    "Korba I": "coal",
+    "Korba III": "coal",
+    "VSTPS-I": "coal",
+    "VSTPS-II": "coal",
+    "VSTPS-III": "coal",
+    "VSTPS-IV": "coal",
+    "VSTPS-V": "coal",
+    "Kawas": "gas",
+    "Gandhar": "gas",
+    "Kakrapar": "nuclear",
+    "Tarapur": "nuclear",
+    "SSP": "hydro",
+    "Sipat I": "coal",
+    "Sipat II": "coal",
+    "RGPPL": "gas",
+    "NSPCL": "coal",
+    "Mauda I": "coal",
+    "Mauda II": "coal",
+    "Sasan": "coal",
+    "CGPL": "coal",
+    "Solapur": "coal",
+    "Gadarwara": "coal",
+    "Lara": "coal",
+    "Khargone": "coal",
+}
+
+EXCHANGES_MAPPING = {
+    "WR-SR": "IN-SO->IN-WE",
+    "WR-ER": "IN-EA->IN-WE",
+    "WR-NR": "IN-NO->IN-WE",
+}
+
+KIND_MAPPING = {
+    "production": {"url": PRODUCTION_URL, "datetime_column": "lastUpdate"},
+    "exchange": {"url": EXCHANGE_URL, "datetime_column": "lastUpdate"},
+    "consumption": {"url": CONSUMPTION_URL, "datetime_column": "current_datetime"},
+}
+
+
+def fetch_data(
+    zone_key: str = "IN-WE",
+    kind: str = None,
```

Comment:
```suggestion
    kind: Optional[str] = None,
```
Otherwise we can't set it to None without breaking the type.